### PR TITLE
Adding types for graphlib-dot

### PR DESCRIPTION
**Pull request by DomParfitt**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/graphlib-dot/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Type definitions for graphlib-dot 0.6
+// Project: https://github.com/dagrejs/graphlib-dot
+// Definitions by: Dom Parfitt <https://github.com/DomParfitt>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { Graph } from 'graphlib';
```
Comment (sheetalkamat) [inline types/graphlib-dot/index.d.ts]: According to https://github.com/dagrejs/graphlib-dot/wiki#graphlibdot-graphlib the graphlibDot is available as browser bundle and hence needs to add global definitions as well. For more information, please refer to https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/graphlib-dot/index.d.ts (hunk 1 above)
Comment (DomParfitt) [reply types/graphlib-dot/index.d.ts]: Thanks for the help. Will do.

Comment (sheetalkamat) [reply types/graphlib-dot/index.d.ts]: You also need YourLibraryName-global.test.ts

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/graphlib-dot/graphlib-dot-tests.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+// Module tests
+let graph: Graph = read('digraph {node 1}');
+
+let graphs: Graph[] = readMany('digraph { node1 }');
+
+let dotStr: string = write(graph);
+
+// Global tests
+graph = graphlibDot.read('digraph {node 1}');
```
Comment (sheetalkamat) [inline types/graphlib-dot/graphlib-dot-tests.ts]: These tests need to be in separate file

File: types/graphlib-dot/test/graphlib-dot-global-tests.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Graph } from "graphlib";
```
Comment (sheetalkamat) [inline types/graphlib-dot/test/graphlib-dot-global-tests.ts]: Do not use imports in global tests

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**